management_api_app/db/repositories/workspace_templates.py

```diff
@@ -9,25 +9,25 @@
 from db.repositories.base import BaseRepository
 from models.domain.resource import ResourceType
 from models.domain.resource_template import ResourceTemplate
-from models.schemas.workspace_template import WorkspaceTemplateInCreate
+from models.schemas.template import TemplateInCreate
 
 
 class WorkspaceTemplateRepository(BaseRepository):
     def __init__(self, client: CosmosClient):
         super().__init__(client, config.STATE_STORE_RESOURCE_TEMPLATES_CONTAINER)
 
     @staticmethod
-    def _workspace_template_by_name_query(name: str) -> str:
-        return f'SELECT * FROM c WHERE c.resourceType = "workspace" AND c.name = "{name}"'
+    def _workspace_template_by_name_query(name: str, resource_type: str = ResourceType.Workspace) -> str:
```

Comment:
nit: This function (and all below) are called **_workspace_template** but this code makes it generic to all templates (workspace_service_template and user_resource_template) - if we want to make it more generic like this we should rename the methods and classes -- but also keeping in mind that some things are specific to the different templates